### ISSUE TITLE
support for sam and vcf format

### ISSUE THREAD

**Issue by lozybean** [biopython/biopython#1467]
[sam format](https://samtools.github.io/hts-specs/SAMv1.pdf) and [vcf format](https://samtools.github.io/hts-specs/VCFv4.2.pdf) is frequently used.
Is it possible to support these two format?

**Comment by peterjc** [comment]
For SAM/BAM, I would recommend pysam which builds on top of the samtools C API: https://github.com/pysam-developers/pysam

However, I did write a pure Python implementation (built on top of ``Bio.bgzf`` for the BAM code) which still exists on my Biopython fork's branches, eg most recently rebased in 2015:

https://github.com/peterjc/biopython/tree/SamBam2015

At the time I wrote that, one of the motivating favours was PyPy's support of C code was limited. See also https://www.biostars.org/p/132289/

If there was interest in using that code of mine, and someone willing to help maintain it, we could look at including it in Biopython. However, it has dated since the SAM/BAM specification has moved on a lot. e.g. Missing CSI indexing, long CIGAR strings in BAM, ...

Also worth a look: https://github.com/quinlan-lab/hts-python

I have not looked at VCF in Python.
